packages/runtime/rollup.config.js

```diff
@@ -1,24 +1,59 @@
-const path = require('path');
-const alias = require('@rollup/plugin-alias');
 const replace = require('@rollup/plugin-replace');
 const copy = require('rollup-plugin-copy');
+const semver = require('semver');
 
 const FEDERATION_DEBUG = process.env.FEDERATION_DEBUG || '';
 
 module.exports = (rollupConfig, projectOptions) => {
   rollupConfig.input = {
     index: 'packages/runtime/src/index.ts',
-    type: 'packages/runtime/src/type/index.ts',
+    types: 'packages/runtime/src/types.ts',
     helpers: 'packages/runtime/src/helpers.ts',
   };
 
   const project = projectOptions.project;
   const pkg = require(project);
+  const RELEASE_NUMBER = Number(
+    `${semver.major(pkg.version)}${semver.minor(pkg.version)}${semver.patch(
+      pkg.version,
+    )}`,
+  );
+
+  if (Number.isNaN(RELEASE_NUMBER)) {
+    throw new Error(
+      `RELEASE_NUMBER is not valid number , please check current pkg.version is valid.`,
+    );
+  }
+  if (rollupConfig.output.format === 'esm' && FEDERATION_DEBUG) {
+    rollupConfig.output.format = 'iife';
+    rollupConfig.output.inlineDynamicImports = true;
+    delete rollupConfig.external;
+    delete rollupConfig.input.type;
+    delete rollupConfig.input.helpers;
+  }
+
+  if (rollupConfig.output.format === 'esm' && FEDERATION_DEBUG) {
```

Comment:
This repeats

Reply:
bad merge ! I will remove the duplicate code